### PR TITLE
:sparkles: call .toString() on non-String headers

### PR DESCRIPTION
**Pull request by techouse**
Enable the user to specify non-String type header values by calling `.toString()` on any non-String Dart type, i.e.

```dart
@Get(path: 'headers')
Future<Response<String>> getHeaders({
  @Header('x-string') String stringHeader,

  /// Using any of the these would have thrown a runtime type error before
  @Header('x-boolean') bool? boolHeader,
  @Header('x-int') int? intHeader,
  @Header('x-double') double? doubleHeader,
  @Header('x-enum') ExampleEnum? enumHeader,
});
```

This will get generated as

```dart
@override
Future<Response<String>> getHeaders({
  required String? stringHeader,
  bool? boolHeader,
  int? intHeader,
  double? doubleHeader,
  ExampleEnum? enumHeader,
}) {
  final Uri $url = Uri.parse('/test/headers');
  final Map<String, String> $headers = {
    if (stringHeader != null) 'x-string': stringHeader, // <-- is a `String`, nothing to do here
    if (boolHeader != null) 'x-boolean': boolHeader.toString(), // <-- calls `.toString()` on a `bool`
    if (intHeader != null) 'x-int': intHeader.toString(), // <-- calls `.toString()` on an `int`
    if (doubleHeader != null) 'x-double': doubleHeader.toString(), // <-- calls `.toString()` on a `double`
    if (enumHeader != null) 'x-enum': enumHeader.toString(), // <-- calls `.toString()` on an `enum`
  };
  final Request $request = Request(
    'GET',
    $url,
    client.baseUrl,
    headers: $headers,
  );
  return client.send<String, String>($request);
}
```

### CODE REVIEW

**Comment by techouse** [comment]
@JEuler @Guldem thoughts?

File: chopper_generator/lib/src/generator.dart [hunk 1]
```diff
@@ -676,18 +676,31 @@ final class ChopperGenerator
...
+            "if ($paramName != null) '$name': $paramName.toString(),",
+          );
+        }
       } else {
-        codeBuffer.writeln('\'$name\': $paramName,');
+        if (parameter.type.isDartCoreString) {
+          codeBuffer.writeln("'$name': $paramName,");
+        } else {
+          codeBuffer.writeln("'$name': $paramName.toString(),");
+        }
```
Comment (Guldem) [inline chopper_generator/lib/src/generator.dart:703]: Small duplication in the code for writing done the heaaders. Maybe could be improved like this:

```dart
      final String headerValue = switch (parameter.type.isDartCoreString) {
        true => "'$name': $paramName,",
        false => "'$name': $paramName.toString(),",
      };
      if (parameter.type.isNullable) {
        codeBuffer.writeln('if ($paramName != null) $headerValue');
      } else {
        codeBuffer.writeln(headerValue);
      }```

**Review by Guldem — COMMENTED** [review]
Nice addition! Got 1 minor feedback thingy.

File: chopper_generator/lib/src/generator.dart (hunk 1 above)
Comment (techouse) [reply chopper_generator/lib/src/generator.dart]: Done

**Review by Guldem — APPROVED** [review]

**Comment by techouse** [comment]
@JEuler bump 😇

**Review by JEuler — APPROVED** [review]
Hmm, looks cool little addition! Sorry for missing it firstly.

**Pull request merged**